### ISSUE TITLE
Sorting in WebUI

### ISSUE THREAD

**Issue by alexanderbock** [OpenSpace/OpenSpace#767]
The sorting of the WebUI Scene properties is not correct;  it's
Solar System
Universe
Milky Way
Other
Shortcuts

and should be
Solar System
Milky Way
Universe
Other
Shortcuts

**Comment by emiax** [comment]
FYI, Before the release I tried implementing this very naively in the `mapStateToProps`, which turned out to be a bad idea performance-wise. I guess we should restructure the redux store somehow, to keep the order in there.

**Comment by alexanderbock** [comment]
Similar issue for the planets as well;  would be good to sort them by distance.  I think I'd prefer to be able to manually specify an order there (which was already implemented for ImGUI through the `openspace.setDefaultGuiSorting` function in `scripts/core_scripts.lua`

**Comment by emiax** [comment]
As of #863, the web gui respects the `Modules.ImGUI.Main.SceneProperties.Ordering` property. Globe browsing layers are sorted by their native order, while everything else is sorted alphabetically. The planet sorting remains though.

**Comment by emiax** [comment]
I suggest waiting with the planetary sorting, and doing that as part of a more general way to sort children of any property owner.

**Comment by micahnyc** [comment]
I added a temp hack for the planetary sorting (before i saw your comment). I think for openspace to communicate this to the gui,  it needs a larger re-work as Emil suggests. For now I just added a dictionary to the api/keys.js in the webgui with entries for 'Planets' and 'Solar System' which can be easily replace by data from the socket when we make it available.

Going to move the issue rather then close it since there is still work to be done.

https://github.com/OpenSpace/OpenSpace-WebGuiFrontend/commit/c25e30058ab5660c31ae5798e962cb22b3589fde
